### DISCUSSION TITLE
Getting started with data science as a High schooler..

### DISCUSSION THREAD

**Discussion by johnyyounas** [community/community#118326] [Other Feature Feedback, Questions, & Ideas]
### Body

**Hey there**, wonderful community! I'm a high school student aspiring to pursue a career in data science. I've heard that data science is an incredibly rewarding and impactful field, playing a crucial role in making important decisions in our modern world. However, I currently only have some basic knowledge of Python. Can anyone provide guidance on a roadmap to learning data science, starting from the basics? I want to ensure that I'm well-prepared for the future and learn in **baby** steps. Thanks in advance for any help!

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by CoderJoe009** [comment]
That's awesome that you're interested in data science! You're already on the right track with Python! Since it's the most popular language for data science, focus on getting really comfortable with it. Also focus on understanding Math concepts like mean, median, standard deviation, statistics , and probability and ,once you've got the coding and math basics down, it's time to dive into data by exploring datasets. I use Google Dataset Search to find datasets (https://datasetsearch.research.google.com/). You should also teach yourself Python Libraries such as Matplotlib or Seaborn for Data Visualization. I also recommend getting onto Kaggle for it is a great platform for anyone interested in Data Science. If you want to do Data Science Courses I recommend DataCamp they offer 3 months free for students through the GitHub Student Developer Pack. his is a great starting point, and there are many resources available online and in libraries to delve deeper. Feel free to ask if you'd like recommendations on specific topics within data science!

**Reply by johnyyounas** [reply]
Thanks for the help CoderJoe009 your help. Also, can you tell me that how to get the student developer pack from git hub; I already have applied in it, but they are not showing any response.
Thanks for your help in advance!

**Comment by MarouaneBenbetka** [comment]
This is the best one : 

https://roadmap.sh/ai-data-scientist

and this is the best book you can read : from zero to hero :
https://www.microsoft.com/en-us/research/uploads/prod/2006/01/Bishop-Pattern-Recognition-and-Machine-Learning-2006.pdf

If this answers your question, please mark it as answered so that others can benefit from it as well. Good luck!

**Comment by DOodle25** [comment]
Python Basics TO Advance
	Refer my public repo for some reference
	-https://github.com/DOodle25/Python-public-
	CS50 python is also good for free
Math Fundamentals
	Introduction to Probability and Data
	Linear Algebra
	Mit opencourseWare have good courses on this
NumPy for numerical computing.
SciPy for scientific computing.

Data Visualization:
	"Data Visualization with Python" on Coursera by the University of Michigan. 
	Matplotlib for basic plotting.
	Seaborn

Machine Learning
	Scikit-learn for implementing machine learning algorithms.
	tenserflow pytorch

Deep Learning (Optional)	
	"Deep Learning Specialization Andrew Ng is the best out there

Data Cleaning and Preprocessing
	Pandas for data cleaning and manipulation.
	Scikit-learn for preprocessing techniques like scaling and encoding.

Model Evaluation and Validation

Big Data Tools (Optional)

Real-World Projects

Continuous Learning.......

remember there is no end to this nor a road map just have to find the things yourself 
i my self am just student so i also know this much hope its helps you

if anyone reading wants to connect please follow me :>  <3

**Reply by johnyyounas** [reply]
Thanks for the help! Marouane Benbetka, appreciate a lot.
